compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -463,6 +464,58 @@ class ResolverImpl(
         return OverridingUtil.overrides(subDescriptor, superDescriptor, true, true)
     }
 
+    private fun isOriginal(original: KSDeclaration, candidate: KSDeclaration): Boolean {
+        incrementalContext.recordLookupForDeclaration(original)
+        incrementalContext.recordLookupForDeclaration(candidate)
+        val originalDescriptor = when (original) {
+            is KSPropertyDeclaration -> resolvePropertyDeclaration(original)
+            is KSFunctionDeclaration -> resolveFunctionDeclaration(original)?.propertyIfAccessor
+            else -> return false
+        }
+
+        val candidateDescriptors = when (candidate) {
+            is KSPropertyDeclaration -> resolvePropertyDeclaration(candidate)?.overriddenDescriptors
+            is KSFunctionDeclaration -> resolveFunctionDeclaration(candidate)?.overriddenDescriptors
```

Comment:
Should we check whether `resolveFunctionDeclaration(candidate)` is a fake override or so? So as to make sure that it is the original. I.e., in case there is a `f1()` defined in `BaseOverride`.

BTW, `FunctionDescriptor.original` sounds like what we need but I'm not sure.

Reply:
original didn't help here, my guess is this is for type substitution, or other purposes.  we might not really want to find the one which is ultimately the original declaration, this function is mainly for filtering out unnecessary checkings, so leaving the check loose is safer than making it more strict.